time_utils/automation/lib/python/community/time_utils.py

```diff
@@ -14,11 +14,18 @@
 limitations under the License.
 """
 import re
-from datetime import datetime, timedelta
+import sys
+import datetime
+from datetime import datetime, date, time, timedelta
 from core.log import logging, LOG_PREFIX
-from core.date import to_joda_datetime
+from core.date import to_joda_datetime, to_python_datetime, to_java_zoneddatetime
+from java.time import ZonedDateTime
+from java.time.temporal import ChronoUnit
 from org.joda.time import DateTime
+from org.eclipse.smarthome.core.library.types import DateTimeType, StringType, DecimalType, PercentType, QuantityType
 from core.jsr223 import scope
+import traceback
+from core.triggers import when
```

Comment:
Why import when? This isn't a script. We can't have rules in here.

Reply:
Oups, that shouldn't be there